tests/test_aps.py

```diff
@@ -11,166 +11,208 @@
-
-    spider = aps_spider.APSSpider()
-    records = list(
-        spider.parse(
-            fake_response_from_file(
-                'aps/aps_single_response.json',
-                response_type=TextResponse,
-            )
+
+    records = []
+
+    for file in ('aps/aps_single_response.json', 'aps/aps_single_response2.json'):
+        fake_response = fake_response_from_file(
+            file,
+            response_type=TextResponse,
         )
-    )
+        spider = aps_spider.APSSpider()
+        records.extend(list(spider.parse(fake_response)))
+
     assert records
+    assert len(records) == 2
+
     return records
 
 
 def test_abstract(results):
     """Test extracting abstract."""
-    abstract = (
+    abstracts = (
         "We use a popular fictional disease, zombies, in order to introduce techniques used in modern epidemiology"
         " modeling, and ideas and techniques used in the numerical study of critical phenomena. We consider variants of"
         " zombie models, from fully connected continuous time dynamics to a full scale exact stochastic dynamic"
         " simulation of a zombie outbreak on the continental United States. Along the way, we offer a closed form"
         " analytical expression for the fully connected differential equation, and demonstrate that the single person"
         " per site two dimensional square lattice version of zombies lies in the percolation universality class. We end"
         " with a quantitative study of the full scale US outbreak, including the average susceptibility of different"
-        " geographical regions."
+        " geographical regions.",
+        ""
     )
-    for record in results:
-        assert 'abstract' in record
-        assert record['abstract'] == abstract
+    for abstract, record in zip(abstracts, results):
+        if abstract:
+            assert 'abstract' in record
+            assert record['abstract'] == abstract
+        else:
+            assert 'abstract' not in record
 
 
 def test_title(results):
     """Test extracting title."""
-    title = "You can run, you can hide: The epidemiology and statistical mechanics of zombies"
-    for record in results:
+    titles = ("You can run, you can hide: The epidemiology and statistical mechanics of zombies",
+              '')
+    for title, record in zip(titles, results):
         assert 'title' in record
         assert record['title'] == title
 
 
 def test_date_published(results):
     """Test extracting date_published."""
-    date_published = "2015-11-02"
-    for record in results:
+    dates_published = ("2015-11-02",
+                       '2015-11-02')
+    for date_published, record in zip(dates_published, results):
         assert 'date_published' in record
         assert record['date_published'] == date_published
 
 
 def test_page_nr(results):
     """Test extracting page_nr"""
-    page_nr = ["11"]
-    for record in results:
+    page_nrs = [["11"],
+                ['0']]
```

Comment:
Pretty good question in fact. I don't know why it was created like this and since I didn't know the reason I didn't want to change it.
However, sometimes it's an array of int, other times array of string, so I'll refactor that.